modules/deployment/variables.tf

```diff
@@ -150,6 +152,24 @@ variable "ecr_repository_name" {
   type        = string
 }
 
+variable "post_deployment_stages" {
+  type = list(object({
+    name = string
+    actions = list(object({
+      name             = string
+      category         = string
+      owner            = string
+      provider         = string
+      version          = string
+      input_artifacts  = list(any)
+      output_artifacts = list(any)
+      configuration    = map(any)
```

Comment:
is this structure always required?

Reply:
I made those optional which are also optional in https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/codepipeline#action. Only works with terraform > 1.0

Reply:
<img width="762" alt="grafik" src="https://github.com/moritzzimmer/terraform-aws-lambda/assets/25341735/e03cd6b4-f31b-4e92-af04-a7b38738ac69">

Hmm. What do think? If the experiment will be deprecated / removed again I think we can migrate this none-breaking by changing this to a map(any) and then using lookup extensivly.

Reply:
I like the approach of using optional with tf 1.3 here